Mathlib/Tactic/TryThis.lean

```diff
@@ -32,14 +32,29 @@ def addHaveSuggestion (ref : Syntax) (t? : Option Expr) (e : Expr) :
       `(tactic| let this := $estx)
   addSuggestion ref tac
 
-/-- Add a suggestion for `rw [h]`. -/
-def addRewriteSuggestion (ref : Syntax) (e : Expr) (symm : Bool) (type? : Option Expr := none) :
+/-- Add a suggestion for `rw [h] at loc`. -/
+def addRewriteSuggestion (ref : Syntax) (e : Expr) (symm : Bool)
+  (type? : Option Expr := none) (loc? : Option Expr := none)
+  (origSpan? : Option Syntax := none) :
     TermElabM Unit := do
   let estx ← delabToRefinableSyntax e
-  let tac ← if symm then `(tactic| rw [← $estx]) else `(tactic| rw [$estx:term])
-  let mut tacMsg := if symm then m!"rw [← {e}]" else m!"rw [{e}]"
+  let tac ←
+    if let some loc := loc? then
+      if symm then
+        `(tactic| rw [← $estx] at $(← delab loc):term)
+      else
+        `(tactic| rw [$estx:term] at $(← delab loc):term)
+    else
+      if symm then `(tactic| rw [← $estx]) else `(tactic| rw [$estx:term])
+
+  let mut tacMsg :=
+    if let some loc := loc? then
+      if symm then m!"rw [← {e}] at {loc}" else m!"rw [{e}] at {loc}" -- TODO fix
```

Comment:
what's the TODO about?

Reply:
I can't remember right now, if there is nothing obviously wrong with it i guess it's ok? If the last commit modifying the line did something other than add the comment then I'd think i just forgot to remove it

Reply:
Also btw @digamma0 if you know how to reduce the nested it's in the tactic quotations above I'd be happy to learn how. I.e. I want to add the at loc part only if needed and the back arrow when needed but the way it's done here is unnecessarily repetitive it seems.

Reply:
why is `loc?` an Expr? Shouldn't it be a `Location`? If you construct an `Option Location` you should be able to interpolate it into the syntax quotation without as many if statements, e.g.
```
    let loc <- loc?.mapM fun loc => `(location| at $(← delab loc):term)
    if symm then `(tactic| rw [← $estx] $(loc)?) else `(tactic| rw [$estx:term] $(loc)?)
```